packages/flutter/lib/src/cupertino/checkbox.dart

```diff
@@ -318,14 +359,19 @@ class _CupertinoCheckboxState extends State<CupertinoCheckbox> with TickerProvid
   @override
   Widget build(BuildContext context) {
     // Colors need to be resolved in selected and non selected states separately.
+    // The `states` getter constructs a new set every time, making it safe to edit in place.
     final Set<WidgetState> activeStates = states..add(WidgetState.selected);
     final Set<WidgetState> inactiveStates = states..remove(WidgetState.selected);
 
     // Since the states getter always makes a new set, make a copy to use
     // throughout the lifecycle of this build method.
     final Set<WidgetState> currentStates = states;
 
-    final Color effectiveActiveColor = _defaultFillColor.resolve(activeStates);
+    final Color effectiveActiveColor = widget.fillColor?.resolve(activeStates)
+        ?? _defaultFillColor.resolve(activeStates);
```

Comment:
Style guide stipulates indenting by two characters: https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#indent-multi-line-argument-and-parameter-lists-by-2-characters

```suggestion
      ?? _defaultFillColor.resolve(activeStates);
```